src/App.tsx

```diff
@@ -1,14 +1,46 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todo, setTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [filterTodoBy, setFilterTodoBy] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then((todosFromServer) => {
+        setTodos(todosFromServer);
+        setIsLoading(true);
+      });
+  }, []);
+
+  const filterTodos = todos
+    .filter((todoItem) => {
+      if (filterTodoBy === 'active') {
+        return !todoItem.completed;
+      }
+
+      if (filterTodoBy === 'completed') {
```

Comment:
Better create enum for filter type. And use it here `const [filterTodoBy, setFilterTodoBy] = useState('all');` too.